haystack/preview/components/samplers/top_p.py

```diff
@@ -0,0 +1,154 @@
+import logging
+from pathlib import Path
+from typing import List, Optional, Union, Dict, Any
+
+import numpy as np
+from haystack.preview import ComponentError
+
+from haystack.lazy_imports import LazyImport
+from haystack.preview import Document, component, default_from_dict, default_to_dict
+
+
+logger = logging.getLogger(__name__)
+
+
+with LazyImport(
+    message="Run 'pip install transformers[torch,sentencepiece]==4.32.1 sentence-transformers>=2.2.0'"
+) as torch_and_transformers_import:
+    from sentence_transformers import CrossEncoder
```

Comment:
Not sure if this is super important, but we were able to remove the `sentence-transformers` dependency and only use `transformers` for running Cross Encoders in the `SentenceTransformersRanker` node. The `CrossEncoder` class in sentence transformers is a thin wrapper around native `transformers` code so it seemed mostly unnecessary and it allowed us to expose more of the loading options and such provided by `transformers`. So I'd consider doing the same here.

Reply:
Ok, will do that optimization later. Good point, thanks @sjrl